kube-core/src/params.rs

```diff
@@ -353,6 +353,35 @@ pub struct DeleteParams {
     pub preconditions: Option<Preconditions>,
 }
 
+impl DeleteParams {
```

Comment:
Do we even want to support leaving `propagation_policy` unset? Maybe we should drop the `Default` impl entirely?

Reply:
I don't have a strong opinion here, but I can see there being cases where folks simply want to use whatever the K8s API server defaults to.

Reply:
I think its fine to keep it optional as per the openapi spec. It has a meaning withing the api-server, so we didn't try to limit what the api can do just because we think it's probably not a good idea on all objects.